pkg/ddb/repository.go

```diff
@@ -212,8 +252,8 @@ func (r *repository) BatchDeleteItems(ctx context.Context, value interface{}) (*
 	})
 }
 
-func (r *repository) batchWriteItem(ctx context.Context, value interface{}, reqBuilder func(map[string]*dynamodb.AttributeValue) *dynamodb.WriteRequest) (*OperationResult, error) {
-	items, err := refl.InterfaceToInterfaceSlice(value)
+func (r *repository) batchWriteItem(ctx context.Context, values []map[string]*dynamodb.AttributeValue, reqBuilder func(map[string]*dynamodb.AttributeValue) *dynamodb.WriteRequest) (*OperationResult, error) {
+	items, err := refl.InterfaceToInterfaceSlice(interface{}(values))
```

Comment:
It doesn't look right to use reflection for this - you already have a slice, it is just of a too specific type. Instead, you could easily change chunk to accept your slice, then you also don't need to cast later

Reply:
That actually never was reflection, but chunking on generic (=interface) level